### PR TITLE
Add docs for RPC Gating process

### PR DESCRIPTION
**Pull request by hughsaunders**
These docs live in the jenkins-rpc repo so that doc changes can be done
alongside code changes. The docs will be linked from the rpc-openstack
readme in a future commit.

Connects rcbops/u-suk-dev#224

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## Contributing
+
+Contributions are welcome, please observe the following guidelines:
+  * All issues are tracked in [waffle](https://waffle.io/rcbops/u-suk-dev), as per the [rpc-openstack](https://github.com/rcbops/rpc-openstack/blob/master/CONTRIBUTING.md) process.
+  * All contributions must be approved by another team member. However each contriubtion only needs one review; the first reviewer can merge if they approve of the change.
```
Comment (major) [inline CONTRIBUTING.md]: `s/contriubtion/contribution`

Comment (hughsaunders) [inline CONTRIBUTING.md]: @major Thanks, fixed.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+## Contributing
+
+Contributions are welcome, please observe the following guidelines:
```
Comment (major) [inline CONTRIBUTING.md]: Should this have a "but"?  "Contributions are welcome, but please observe..."

Comment (hughsaunders) [inline CONTRIBUTING.md]: Done, thanks.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+  * All issues are tracked in [waffle](https://waffle.io/rcbops/u-suk-dev), as per the [rpc-openstack](https://github.com/rcbops/rpc-openstack/blob/master/CONTRIBUTING.md) process.
+  * All contributions must be approved by another team member. However each contribution only needs one review; the first reviewer can merge if they approve of the change.
+  * Changes must update the docs if appropriate.
```
Comment (hughsaunders) [inline CONTRIBUTING.md]: Done, thanks.

File: README.md [hunk 4]
```diff
@@ -1,4 +1,105 @@
...
-=========
+# RPC-Openstack Gating
+<img src="fire-jenkins.png" width="300">
 
-A set of ansible plays and dockerfiles that will allow rpc to automate deployment testing across mutliple environments
+## Linting
+Linting is checking the syntax and style of the code without executing it. RPC Lint jobs are run by [Travis CI](https://travis-ci.org/).
+
+### Linting Job Mechanics
+  1. [rcbops/rpc-openstack](https://github.com/rcbops/rpc-openstack) has the travis integration enabled. This creates a webhook that posts [PR](https://github.com/rcbops/rpc-openstack/pulls) notifications to Travis. Note that the webhook isn't visible in the [github ui](https://github.com/rcbops/rpc-openstack/settings/hooks), as there is an official integration.
```
Comment (hughsaunders) [inline README.md]: Yeah, GFM renderer will replace the ones with the actual numbers.

File: README.md [hunk 5]
```diff
@@ -1,4 +1,105 @@
...
+      1. [flake8](https://gitlab.com/pycqa/flake8) which runs      [scripts/linting-pep8.sh](https://github.com/rcbops/rpc-openstack/blob/master/scripts/linting-pep8.sh) to check for ansible style violations.
+      1. [ansible-lint](https://github.com/willthames/ansible-lint) which runs [scripts/linting-ansible.sh](https://github.com/rcbops/rpc-openstack/blob/master/scripts/linting-pep8.sh). This script runs flake8 over all files with a python [shebang](https://en.wikipedia.org/wiki/Shebang_(Unix)) to check for python style violations.
+  1. Once travis has completed it posts results back to github via the API, and github displays them in the PR.
+
+## Functional Tests
+Functional tests are defined using [Jenkins Job Builder](http://docs.openstack.org/infra/jenkins-job-builder/) in the [jenkins-rpc](https://github.com/rcbops/jenkins-rpc/tree/master/rpc-jobs) repo.
+Jenkins jobs builder translates YAML job descriptions into Jenkins job configuration. This allows job config to be stored in a repo in a readable format.
+
+Note that some Jobs have not yet been migrated to JJB and are defined in Jenkins directly.
```
Comment (hughsaunders) [inline README.md]: Done, thanks.

File: README.md [hunk 6]
```diff
@@ -1,4 +1,105 @@
...
+### Jenkins Job Mechanics
+  1. [rcbops/rpc-openstack](https://github.com/rcbops/rpc-openstack) has a webhook configured to notify Jenkins of new/updated PRs.
+  1. Jenkins has the [github pull request builder (ghprb)](https://wiki.jenkins-ci.org/display/JENKINS/GitHub+pull+request+builder+plugin) plugin enabled. This interprets the received webhook and kicks off the necessary job.
+  1. Once the job has completed, jenkins posts the results back to github via the API, and github displays them in the PR.
+
+### Jenkins Jobs
+
+###### JJB-RPC-AIO-*
+These jobs are all instances of the same JJB [Job Template](https://github.com/rcbops/jenkins-rpc/blob/master/rpc-jobs/jobs.yaml#L98). They are instantiated using a [project](http://docs.openstack.org/infra/jenkins-job-builder/definition.html#project), which is currently 2x4 matrix (swift,ceph)x(kilo,liberty,mitaka,master)  See [JJB-RPC-AIO_master-swift](http://jenkins.propter.net/job/JJB-RPC-AIO_master-swift/) as an example.
```
Comment (hughsaunders) [inline README.md]: Done, thanks.

**Pull request merged**